src/main/java/org/candlepin/util/ConsumerFeedFactExtractor.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2009 - 2025 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+
+package org.candlepin.util;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+public class ConsumerFeedFactExtractor {
+
+    private ConsumerFeedFactExtractor() {
+        // Intentionally left empty
+    }
+
+    // List of allowed fact keys (exact matches)
+    private static final Set<String> ALLOWED_FACTS = Set.of(
+        "cpu.cpu(s)",
```

Comment:
nit: if we're going to have hardcoded lists of facts, we should sort them so adding new entries can be done in a way that reduces the chance for dupes, and it's easier to quickly identify if a given value is in it.